### PR TITLE
Configurable caching for negative results (NXDOMAIN) (#287)

### LINKED ISSUE #287: Do not cache nxdomain
Hi 👋🏼 

It would be cool if this could be default, or configurable. Usually I'm testing things in my homelab and my service isn't online just yet when I try to hit it. Blocky caches that nxdomain so I always need to bounce the container to clear the cache.

Thanks!

### PR DESCRIPTION
**Pull request by kwitsch**
Changes:

- added cache.cacheTimeNegative (time.Duration)
- if not configured the default cache time stays at 30 min
- if set to a value below zero caching of negative responses is disabled

closes #287

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 bin/
 .idea
 .github
-testdata/
```
Comment (0xERR0R) [inline .dockerignore:4]: docker build doesn't perform tests, therefore this directory is excluded to minimize the amount of transferred data to the docker daemon. Did you need this directory for your build?

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,4 @@ todo.txt
 !docs/config.yml
 node_modules
 package-lock.json
+Dockerfile.Test
```
Comment (0xERR0R) [inline .gitignore:16]: I think it is something from your local tests?

File: config/config.go [hunk 3]
```diff
@@ -324,6 +324,7 @@ type ClientLookupConfig struct {
 type CachingConfig struct {
 	MinCachingTime        Duration `yaml:"minTime"`
 	MaxCachingTime        Duration `yaml:"maxTime"`
+	CacheTimeNegative     Duration `yaml:"cacheTimeNegative" default:"30m"`
```
Comment (0xERR0R) [inline config/config.go:327]: :+1: Thanks for adoption of "defaults"

File: go.mod [hunk 4]
```diff
@@ -20,7 +19,6 @@ require (
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/cobra v1.2.1
 	github.com/stretchr/testify v1.7.0
-	github.com/swaggo/swag v1.7.3
```
Comment (0xERR0R) [inline go.mod:23]: I think you need to run "go mod tidy" and commit the changes, otherwise "make all" won't run

**Review by 0xERR0R — CHANGES_REQUESTED** [review]
Looks good for me, please run "go mod tidy" and commit the changes, I think there is something wrong

File: .gitignore (hunk 2 above)
Comment (kwitsch) [reply .gitignore:16]: Yes

File: .dockerignore (hunk 1 above)
Comment (kwitsch) [reply .dockerignore:4]: Needed it for local testing (Dockerfile.Test)

**Comment by kwitsch** [comment]
> Looks good for me, please run "go mod tidy" and commit the changes, I think there is something wrong

Sorry currently not at home.
I can commit it tomorrow.

**Comment by 0xERR0R** [comment]
Thanks for your work, I had some spare time today and I think we can merge it now

**Review by 0xERR0R — APPROVED** [review]

**Pull request merged**